### ISSUE TITLE
Gifted Chat breaks and app crashes

### ISSUE THREAD

**Issue by LMestre14** [FaridSafi/react-native-gifted-chat#674]
#### Issue Description

Everytime I scroll the rendered messages by gifted chat, the performance is going down exponentially by the point that the app crashes

#### Steps to Reproduce / Code Snippets

I only installed the component, and put it on render and sending via state the messages that I receive from socket.
I've used the component before in another project, same story and structure (other project had gifted chat version ^0.2.7 and react-native was 0.46.4)

#### Expected Results

What I wanted to happen is that the scroll first of all would be smooth, I can see the scroll like "rendering" piece by piece the messages

#### Additional Information

The best I could do to improve performance for Android, since IOS is okay, was adding resizeMethod='resize' to GiftedAvatar, and yet the scroll isn't improving and the app crashes after a few times of scrolling. Also made all the rendered components PureComponent, since they only show information, and yet that didn't work.

I've noticed that GiftedChat is still using ListView to render the list of messages, but since the other project also had GiftedChat with ListView I'm not sure if it's the main problem of performance and crashing

* React version: 16.0.0-beta.5
* React Native version: ^0.49.3
* react-native-gifted-chat version: ^0.3.0
* Platform(s) (iOS, Android, or both?): Android only

**Comment by LMestre14** [comment]
**Additional Information:**

I've tested this in multiple devices, in all of them (debug or apk) works well, except on the company's smartphone (Huawei VNS-L31).

In all the other devices, some were slow others were pretty fast, the gifted chat worked like a charm, except on that specific device

**Comment by xcarpentier** [comment]
Hello,
Can you please provide some logs?
Thank you

**Comment by hugoh1995** [comment]
Hi I have the same issue on a Samsung. Here are my logs:

Hi, I have no idea why but my app crashes on multiple occasions on a real android device, can anyone help me figure out how to solve this?

It crashes right after **I touch the screen with my finger** on this specific view. What could cause a touch issue to crash?

This view is using react-native-gifted-chat and I suspect the error is linked to that library.

   
    ```
I/InputReader( 3064): Touch event's action is 0x0 (deviceType=0) [pCnt=1, s=0.1314 ] when=15389736722000
    I/InputDispatcher( 3064): Delivering touch to (3262): action: 0x4, toolType: 1
    I/InputDispatcher( 3064): Delivering touch to (7658): action: 0x0, toolType: 1
    D/ViewRootImpl( 7658): ViewPostImeInputStage ACTION_DOWN
    D/SSRM:n  ( 3064): SIOP:: AP = 370
    I/ActivityManager( 3064): Process org.simalliance.openmobileapi.service:remote (pid 7833)(adj 1) has died(26,52)
    W/ActivityManager( 3064): Scheduling restart of crashed service org.simalliance.openmobileapi.service/.SmartcardService in 253119ms
    I/ActivityManager( 3064): Process com.sec.android.inputmethod (pid 7430)(adj 1) has died(14,52)
    W/ActivityManager( 3064): Scheduling restart of crashed service com.sec.android.inputmethod/com.diotek.ime.framework.spellcheckservice.SamsungIMESpellCheckerService in 263095ms
    W/ActivityManager( 3064): Scheduling restart of crashed service com.sec.android.inputmethod/.SamsungKeypad in 273095ms
    W/ActivityManager( 3064): Scheduling restart of crashed service com.sec.android.inputmethod/com.nuance.connect.service.ConnectClient in 283095ms
    W/InputMethodManagerService( 3064): Session failed to close due to remote exception
    W/InputMethodManagerService( 3064): android.os.DeadObjectException
    W/InputMethodManagerService( 3064): 	at android.os.BinderProxy.transactNative(Native Method)
    W/InputMethodManagerService( 3064): 	at android.os.BinderProxy.transact(Binder.java:496)
    W/InputMethodManagerService( 3064): 	at com.android.internal.view.IInputMethodSession$Stub$Proxy.finishSession(IInputMethodSession.java:305)
    W/InputMethodManagerService( 3064): 	at com.android.server.InputMethodManagerService.finishSessionLocked(InputMethodManagerService.java:2067)
    W/InputMethodManagerService( 3064): 	at com.android.server.InputMethodManagerService.clearClientSessionLocked(InputMethodManagerService.java:2058)
    W/InputMethodManagerService( 3064): 	at com.android.server.InputMethodManagerService.clearCurMethodLocked(InputMethodManagerService.java:2084)
    W/InputMethodManagerService( 3064): 	at com.android.server.InputMethodManagerService.onServiceDisconnected(InputMethodManagerService.java:2103)
    W/InputMethodManagerService( 3064): 	at android.app.LoadedApk$ServiceDispatcher.doDeath(LoadedApk.java:1278)
    W/InputMethodManagerService( 3064): 	at android.app.LoadedApk$ServiceDispatcher$RunConnection.run(LoadedApk.java:1292)
    W/InputMethodManagerService( 3064): 	at android.os.Handler.handleCallback(Handler.java:739)
    W/InputMethodManagerService( 3064): 	at android.os.Handler.dispatchMessage(Handler.java:95)
    W/InputMethodManagerService( 3064): 	at android.os.Looper.loop(Looper.java:135)
    W/InputMethodManagerService( 3064): 	at com.android.server.SystemServer.run(SystemServer.java:686)
    W/InputMethodManagerService( 3064): 	at com.android.server.SystemServer.main(SystemServer.java:468)
    W/InputMethodManagerService( 3064): 	at java.lang.reflect.Method.invoke(Native Method)
    W/InputMethodManagerService( 3064): 	at java.lang.reflect.Method.invoke(Method.java:372)
    W/InputMethodManagerService( 3064): 	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1405)
    W/InputMethodManagerService( 3064): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1200)
    W/InputDispatcher( 3064): channel '2d254075 com.android.systemui.ImageWallpaper (server)' ~ Consumer closed input channel or an error occurred.  events=0x9
    E/InputDispatcher( 3064): channel '2d254075 com.android.systemui.ImageWallpaper (server)' ~ Channel is unrecoverably broken and will be disposed!
    I/ActivityManager( 3064): Process com.android.systemui.imagewallpaper (pid 7400)(adj 1) has died(133,39)
    W/ActivityManager( 3064): Scheduling restart of crashed service com.android.systemui/.ImageWallpaper in 0ms
    W/InputDispatcher( 3064): Attempted to unregister already unregistered input channel '2d254075 com.android.systemui.ImageWallpaper (server)'
    I/WindowState( 3064): WIN DEATH: Window{2d254075 u0 com.android.systemui.ImageWallpaper}
    I/StatusBar( 3262): Icon Only
    D/PanelView( 3262): There is/are notification(s) 
    W/InputDispatcher( 3064): channel '105f1974 com.dogtimereactinit/com.dogtimereactinit.MainActivity (server)' ~ Consumer closed input channel or an error occurred.  events=0xd
    E/InputDispatcher( 3064): channel '105f1974 com.dogtimereactinit/com.dogtimereactinit.MainActivity (server)' ~ Channel is unrecoverably broken and will be disposed!
    I/Sensors ( 3064): AccelerometerSensor mEnabled 1, handle 0, en 0
    D/InputDispatcher( 3064): Waiting for application to become ready for input: 7658.  Reason: Waiting because the touched window's input channel is not registered with the input dispatcher.  The window may be in the process of being removed
```

**Comment by alexgnatrow** [comment]
I had the same issue. In my case it was really silly mistake in my styled component like:
  `background: ${props => props.hex} no-repeat; `
when really I needed to secure this 
` background: ${props => props.hex || '#fff'} no-repeat; ` 
or even
` background: ${props => props.hex && props.hex || '#fff'} no-repeat; `
I have no idea where this could cause performance hit or even crashing in gifted chat, but thats basically it. I crawled all over the adb logs before I realized that it is easier to just copy working branch component into current and try. Maybe this would be helpful for someone.
Btw thanks for a great lib!
